docs/trace/enriching-activity-processor/README.md

```diff
@@ -0,0 +1,118 @@
+# EnrichingActivityProcessor
+
+Sometimes the built-in instrumentation doesn't add enough data or you want to
+augment the data it provides with contextual information your application or
+library has available. For these cases OpenTelemetry .NET provides an
+`EnrichingActivityProcessor` and `EnrichmentScope` API.
+
+## Basic Usage
+
+### Add EnrichingActivityProcessor
+
+To enrich your `Activity` objects first add the `EnrichingActivityProcessor` to
+your `TracerProvider`:
+
+```csharp
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddHttpClientInstrumentation()
+    .AddProcessor(new EnrichingActivityProcessor())
+    .AddConsoleExporter()
+    .Build();
+```
+
+**Note:** Order is important. Make sure your `EnrichingActivityProcessor` is
+added before your `Exporter` and any other `ActivityProcessor`s you are using.
```

Comment:
Not suggesting any action for this PR, but food for thought...

This note makes me wonder if we should consider codifying processor/exporter ordering in an effort to make this less prone to messing up. I don't have a clear suggestion at the moment, but maybe a combination of "type" and "priority".

Like if we know it's an "exporter type" versus "processor type", can the builder be smart enough to make sure to place it at the end?

Furthermore, maybe not all processors are equal. Some may be "high priority" and should therefore be injected at the beginning of the pipeline. This would in turn require some logic to deal with ties.

Reply:
@reyang Tagging you just in case you didn't see this. Some food for thought as we improve/refactor the build-up. The ordering problem is similar to [ASP.NET Core Middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.1#middleware-order). Not much provided there, other than guidance that order is critical.

Interesting ideas about distinguishing "exporting processors" versus "enriching processors" versus other types.

Reply:
Reminded me of [TSR](https://en.wikipedia.org/wiki/Terminate_and_stay_resident_program) and [IRP](https://www.microsoftpressstore.com/articles/article.aspx?p=2201309&seqNum=3).